### PR TITLE
chore(api): mark meshgateway as non-policy resource

### LINKED ISSUE #9268: Remove `MeshGateway` objects from /policies
### Description

Change the `/policies` API such that it doesn’t include the policy type definition for `MeshGateway` objects anymore.

Related to https://github.com/kumahq/kuma-gui/issues/2185.

We’re changing the UI to have a dedicated Gateways tab which includes two views:

- Built-in Gateways (lists `MeshGateway`s)
- Delegated Gateways (lists `ServiceInsight` objects of service type `gateway_delegated`)

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
### Checklist prior to review

Fix #9268

This PR removes MeshGateway from policies endpoint, it's not marked as non policy. This also means that we cannot do inspect API on this endpoint `/meshes/{mesh}/meshgateways/{policyName}/_resources/dataplanes`, but it's fine since we don't need this anymore in the GUI.

- [X] [Link to relevant issue][1] as well as docs and UI issues --
- [X] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/core/resources/apis/mesh/zz_generated.resources.go [hunk 1]
```diff
@@ -1024,7 +1024,7 @@ var MeshGatewayResourceTypeDescriptor = model.ResourceTypeDescriptor{
 	KumactlArg:          "meshgateway",
 	KumactlListArg:      "meshgateways",
 	AllowToInspect:      false,
-	IsPolicy:            true,
+	IsPolicy:            false,
```
Comment (michaelbeaumont) [inline pkg/core/resources/apis/mesh/zz_generated.resources.go:1027]: Might be worth mentioning in UPGRADE.md since it's an API change?

**Review by michaelbeaumont — APPROVED** [review]
Does this affect `kumactl`?

**Comment by jakubdyszkiewicz** [comment]
it does not affect kumactl because
```
var MeshGatewayResourceTypeDescriptor = model.ResourceTypeDescriptor{
	Name:                MeshGatewayType,
	Resource:            NewMeshGatewayResource(),
	ResourceList:        &MeshGatewayResourceList{},
	ReadOnly:            false,
	AdminOnly:           false,
	Scope:               model.ScopeMesh,
	KDSFlags:            model.ZoneToGlobalFlag | model.GlobalToAllZonesFlag,
	WsPath:              "meshgateways",
	KumactlArg:          "meshgateway",
	KumactlListArg:      "meshgateways",
	AllowToInspect:      false, # <- here
	IsPolicy:            false,
	SingularDisplayName: "Mesh Gateway",
	PluralDisplayName:   "Mesh Gateways",
	IsExperimental:      false,
}
```

File: pkg/core/resources/apis/mesh/zz_generated.resources.go (hunk 1 above)
Comment (jakubdyszkiewicz) [reply pkg/core/resources/apis/mesh/zz_generated.resources.go:1027]: since it's not used in GUI nor in kumactl, I don't think it's worth it

**Pull request merged**